hw3/01-characters.js

```diff
@@ -1,4 +1,75 @@
-// url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
 
-fetch(url);
+let list = document.querySelector("#list");
+
+const fetchData = (url) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      resolve(data);
+    } catch (error) {
+      reject(error);
+    }
+  });
+};
+
+const createCharacterElements = (character) => {
+  let col = document.createElement("div");
+  col.classList.add("col-3", "p-0", "m-2");
+
+  let figure = document.createElement("figure");
+  figure.classList.add("figure", "h-100");
+
+  figure.addEventListener("mouseover", () => {
+    figure.style.backgroundColor = "darkBlue";
+    figure.style.color = "white";
+  });
+
+  figure.addEventListener("mouseout", () => {
+    figure.style.backgroundColor = "";
+    figure.style.color = "";
+  });
+
+  let img = document.createElement("img");
+  img.classList.add("figure-img", "img-fluid");
+  img.src = character.imageUrl;
+  img.alt = character.image;
+
+  let figcaption = document.createElement("figcaption");
+  figcaption.classList.add("fw-bold", "p-2", "h-25", "d-inline-block");
+  figcaption.innerHTML = character.fullName + "<br>" + character.title;
+
+  figure.appendChild(img);
+  figure.appendChild(figcaption);
+  col.appendChild(figure);
+
+  return col;
+};
+
+const createRows = (characters) => {
```

Comment:
Nice clean function to add characters to page, would be helpful for some comments to describe the this and fetchData() below a little but the code itself looks really good.